evalml/objectives/standard_metrics.py

```diff
@@ -494,6 +522,7 @@ class R2(RegressionObjective):
     score_needs_proba = False
     perfect_score = 1
     is_bounded_like_percentage = False  # Range (-Inf, 1]
+    expected_range = [-1, 1]
```

Comment:
@ParthivNaresh That's a good point! When there are outliers, this will likely raise a high variance warning more often, but I think that makes sense to do as well. In scenarios where each fold can get different enough variation of data where the models perform fairly differently, raising a high variance warning would be beneficial